src/firefox/index.js

```diff
@@ -49,28 +50,33 @@ export type RemotePortFinderFn =
 export function defaultRemotePortFinder(
   {
     portToTry=REMOTE_PORT,
+    retriesLeft=10,
     connectToFirefox=defaultFirefoxConnector,
-  }: RemotePortFinderParams = {}
+  }: RemotePortFinderParams = {},
+
 ): Promise<number> {
   log.debug(`Checking if remote Firefox port ${portToTry} is available`);
-
-  return connectToFirefox(portToTry)
-    .then((client) => {
-      log.debug(`Remote Firefox port ${portToTry} is in use`);
-      client.disconnect();
-      // TODO: instead of throw an error, pick a new random port until
-      // one of them is available.
-      // https://github.com/mozilla/web-ext/issues/283
-      throw new WebExtError(
-        `Cannot listen on port ${portToTry} because it's in use`);
-    })
-    .catch(onlyErrorsWithCode('ECONNREFUSED', () => {
-      // The connection was refused so this port is good to use.
-      return portToTry;
-    }));
+  function tryToFindAnOpenPort() {
+    return connectToFirefox(portToTry)
+      .then((client) => {
+        log.debug(`Remote Firefox port ${portToTry} is in use ` +
+          `(retries remaining: ${retriesLeft} )`);
+        client.disconnect();
+        portToTry++;
+        if (retriesLeft > 0) {
+          retriesLeft--;
+          return tryToFindAnOpenPort();
+        }
+        else {
+          throw new WebExtError('Too many retries on port search');
+        }
+      })
+     .catch(onlyErrorsWithCode('ECONNREFUSED', () => {
```

Comment:
I would like to see the original comment put back here:

    // The connection was refused so this port is good to use.

The reason is that it's confusing to look at such an obtuse error code and see the recovery from it as a successful result. The comment explains why.